### ISSUE TITLE
Factor out XEB fidelity estimator out of XEB experiment code

### ISSUE THREAD

**Issue by viathor** [quantumlib/Cirq#2113]
XEB experiment code is monolithic. In particular, it tightly couples generation of random circuits and fidelity estimation. The latter is potentially useful on its own (e.g. when RQCs have been generated and executed offline), so should be factored out for re-use.

Also, there is more than one formula for estimating fidelity, e.g. on a few qubits one can compute cross-entropies directly while on many qubits one has to resort to formulas based on some assumptions about the shape of the output distribution. Therefore, it makes sense to implement fidelity estimation separate from RQC generation and overall XEB orchestration.

**Comment by mpharrigan** [comment]
See https://github.com/quantumlib/ReCirq/issues/85

**Comment by MichaelBroughton** [comment]
Is this still an issue @viathor, @mpharrigan  ? Is there a plan here ?

**Comment by augustehirth** [comment]
I agree with this and am glad to work on it if there's consensus, as I'm already working on some other XEB tutorial related improvements.

**Comment by dstrain115** [comment]
@NoureldinYosri refactored this code and it is now modular.  This is now resolved.
